src/lavinmq/deduplication.cr

```diff
@@ -0,0 +1,71 @@
+module LavinMQ
+  module Deduplication
+    abstract class Cache(T)
+      abstract def contains?(key : T) : Bool
+      abstract def insert(key : T, ttl : UInt32?)
+    end
+
+    class MemoryCache(T) < Cache(T)
+      def initialize(@size : UInt32)
+        @store = Hash(T, Time::Span?).new
+      end
+
+      def contains?(key : T) : Bool
+        return false unless @store.has_key?(key)
+        ttd = @store[key]
+        return true unless ttd
+        return true if ttd > RoughTime.monotonic
+        @store.delete(key)
+        false
+      end
+
+      def insert(key : T, ttl : UInt32? = nil)
+        @store.shift if @store.size >= @size
+        val = if ttl
+                RoughTime.monotonic + Time::Span.new(nanoseconds: ttl * 1_000_000)
```

Comment:
```suggestion
                RoughTime.monotonic + ttl.seconds.total_nanoseconds
```

Reply:
Did this `val = ttl.try { |v| RoughTime.monotonic + v.milliseconds }`  instead